File: final-evaluation-generated-code/test-3/deterministic.js
import { EventSource } from 'eventsource';

const URL = "http://localhost:3000/http-express-calculator-simple/events/update";

function subscribeeventUpdate() {
    const eventSource = new EventSource(URL);

    eventSource.onopen = () => {
        console.log("Connected Successfully");
    };

    eventSource.onmessage = (e) => {
        try {
            console.log("update: ", JSON.parse(e.data));
        } catch (error) {
            console.error('Error with update:', error);
        }
    };

    eventSource.onerror = (error) => {
        console.error('Error with update:', error);
    };

    return {
        unsubscribe: () => {
            eventSource.close();
        }
    };
}

const event = subscribeeventUpdate();

//Disconnect after 10 seconds
setTimeout(() => {
    event.unsubscribe();
    console.log("Disconnected");
}, 10000);

File: final-evaluation-generated-code/test-9/deterministic.py
import requests

URL = "http://localhost:3000/http-express-calculator-simple/events/update"

def subscribeevent_update():
  while True:
    try: 
      response = requests.get(URL, stream=True, timeout=10)

      if response.status_code == 200:
        for line in response.iter_lines():
          if line:
            data = line.decode("utf-8")
            print(data)
      else:
        print(f"Failed to subscribeevent update. Status code: {response.status_code}, Response: {response.text}")
        break

    # Closing the connection after timeout or on error
    except requests.exceptions.RequestException as error:
      print("Error during the request:", error)
      break

subscribeevent_update()